test/test_device_radix_sort.cu

```diff
@@ -704,7 +704,7 @@ void InitializeSolution(
         // Generate a random permutation from the summary. Such a complicated
         // approach is used to permute the array and compute ranks in a
         // cache-friendly way and in a short time.
-        h_reference_ranks = new NumItemsT[num_items];
+        if (WANT_RANKS) h_reference_ranks = new NumItemsT[num_items];
```

Comment:
Style nitpick -- put conditional bodies on a new line with explicit brackets. We've found some bugs resulting from folks incorrectly modifying this compact conditional style.

```suggestion
        if (WANT_RANKS)
        {
            h_reference_ranks = new NumItemsT[num_items];
        }
```